### DISCUSSION TITLE
Question about installing r2decompiler

### DISCUSSION THREAD

**Discussion by AirShoot125** [radareorg/radare2#21229] [Q&A]
Hi to everyone! [Screenshot_20230105_124923.jpg](https://user-images.githubusercontent.com/114504779/210764241-cd82f4b7-472f-45d9-b5d6-72332fc8ca48.jpg)
I wanted to install r2dec in termux, but it gives an error, how to solve it? I'm beginner in this, so i didn't understand why it was

**Comment by apkunpacker** [comment]
@AirShoot125 Hi can you try updating your r2 version from git instead of using packaged one from termux?
you can follow these step which might help you
```sh
pkg uninstall radare2
```
We removing packaged radare2 from above command as we building from git next. After this step follow these commands

```sh
pkg install build-essential wget git python binutils
cd /data/data/com.termux/files/home
git clone https://github.com/radareorg/radare2
cd radare2
sh sys/termux.sh
```
After installation complete, you can use
```sh
r2pm -U
r2pm -ci r2dec
```

**Comment by trufae** [comment]
There are bugs fixed in git that improve the error messages . The real bug is probably that you are missing a pkg to install in your termux like missing gcc or missing git

**Comment by trufae** [comment]
@apkunpacker maybe we can show this oneliner of apt in r2pm when a missing dep is found. That will make it easier

**Comment by trufae** [comment]
should work fine now afaik
